semgrep-core/src/parsing/tree_sitter/Parse_hack_tree_sitter.ml

```diff
@@ -488,1277 +598,1603 @@ and anon_choice_exp_rep_COMMA_choice_exp_opt_COMMA_e4364bb (env : env)
       let v1 = expression env v1 in
-      let v2 = token env v2 (* "<<=" *) in
+      let v2, op = (* "<<=" *) (token env v2, G.LSL) in
       let v3 = expression env v3 in
-      todo env (v1, v2, v3)
+      G.AssignOp (v1, (op, v2), v3) |> G.e
   | `Exp_GTGTEQ_exp (v1, v2, v3) ->
       let v1 = expression env v1 in
-      let v2 = token env v2 (* ">>=" *) in
+      let v2, op = (* ">>=" *) (token env v2, G.LSR) in
       let v3 = expression env v3 in
-      todo env (v1, v2, v3)
+      G.AssignOp (v1, (op, v2), v3) |> G.e
   | `Exp_PLUSEQ_exp (v1, v2, v3) ->
       let v1 = expression env v1 in
-      let v2 = token env v2 (* "+=" *) in
+      let v2, op = (* "+=" *) (token env v2, G.Plus) in
       let v3 = expression env v3 in
-      todo env (v1, v2, v3)
+      G.AssignOp (v1, (op, v2), v3) |> G.e
   | `Exp_DASHEQ_exp (v1, v2, v3) ->
       let v1 = expression env v1 in
-      let v2 = token env v2 (* "-=" *) in
+      let v2, op = (* "-=" *) (token env v2, G.Minus) in
       let v3 = expression env v3 in
-      todo env (v1, v2, v3)
+      G.AssignOp (v1, (op, v2), v3) |> G.e
   | `Exp_STAREQ_exp (v1, v2, v3) ->
       let v1 = expression env v1 in
-      let v2 = token env v2 (* "*=" *) in
+      let v2, op = (* "*=" *) (token env v2, G.Mult) in
       let v3 = expression env v3 in
-      todo env (v1, v2, v3)
+      G.AssignOp (v1, (op, v2), v3) |> G.e
   | `Exp_SLASHEQ_exp (v1, v2, v3) ->
       let v1 = expression env v1 in
-      let v2 = token env v2 (* "/=" *) in
+      let v2, op = (* "/=" *) (token env v2, G.Div) in
       let v3 = expression env v3 in
-      todo env (v1, v2, v3)
+      G.AssignOp (v1, (op, v2), v3) |> G.e
   | `Exp_PERCEQ_exp (v1, v2, v3) ->
       let v1 = expression env v1 in
-      let v2 = token env v2 (* "%=" *) in
+      let v2, op = (* "%=" *) (token env v2, G.Mod) in
       let v3 = expression env v3 in
-      todo env (v1, v2, v3)
+      G.AssignOp (v1, (op, v2), v3) |> G.e
   | `Exp_STARSTAREQ_exp (v1, v2, v3) ->
       let v1 = expression env v1 in
-      let v2 = token env v2 (* "**=" *) in
+      let v2, op = (* "**=" *) (token env v2, G.Pow) in
       let v3 = expression env v3 in
-      todo env (v1, v2, v3)
+      G.AssignOp (v1, (op, v2), v3) |> G.e
+
+and braced_expression (env : env) ((v1, v2, v3) : CST.braced_expression) =
+  let v1 = (* "{" *) token env v1 in
+  let v2 = expression env v2 in
+  let v3 = (* "}" *) token env v3 in
+  (v1, v2, v3)
 
 and call_expression (env : env) ((v1, v2, v3) : CST.call_expression) =
   let v1 =
     match v1 with
     | `Exp x -> expression env x
-    | `Choice_array x -> collection_type env x
-  in
-  let v2 =
-    match v2 with Some x -> type_arguments env x | None -> todo env ()
+    | `Choice_array x ->
+        G.N (G.Id (collection_type env x, G.empty_id_info ())) |> G.e
   in
+  let v2 = match v2 with Some x -> type_arguments env x | None -> None in
   let v3 = arguments env v3 in
-  todo env (v1, v2, v3)
+  G.Call (v1, v3) |> G.e
```

Comment:
Like I mentioned to @aryx the other day, maintenance is much easier if the left-handside variables are given somewhat meaningful names instead of `v1`, `v2` etc. Instead of this:
```ocaml
let v2 = match v2 with Some x -> type_arguments env x | None -> None in
```
I write this:
```ocaml
let type_args = match v2 with Some x -> type_arguments env x | None -> None in
```
It's easy to give reasonable names while their meaning fresh in your mind but it's much harder later.

Here, the return value would be something like
```ocaml
G.Call (f, args) |> G.e
```
so when v3 becomes v4 after some grammar change, it helps reduce confusion.

I've started using [this style systematically for the bash parser](https://github.com/returntocorp/semgrep/blob/47d0935186b25fc9e9c23c1db229a59756f512ea/semgrep-core/src/parsing/tree_sitter/Parse_bash_tree_sitter.ml#L362-L366), after experiencing painful upgrades with javascript and typescript.

Reply:
Noted! I'll update as I go with future changes. I'll also try to rename the `anon_choice_` functions to be more descriptive.